CONTRIBUTING.md

```diff
@@ -0,0 +1,46 @@
+# Contributing
+
+## Forward
+
+So, you're looking to contribute to Fabric API. Well, you've come to the right place. This guide is dedicated to teaching you everything you need to know about contributing to Fabric API. This guide might be kind of lengthy, but there's a lot of useful information in it, so make sure to read until the end.
+
+Before we begin, some friendly advice:
+1. We don't tend to ban people just for having ideas, no matter how outlandish. However, “no” can mean “no”, and un-constructive responses to constructive arguments can get on our nerves after some time. We are generally unpaid volunteers who dedicate a lot of free time to the tiring job of social interaction - please don't make our work harder than it already is.
+2. We're here to help each other learn, after all. Don't hesitate to talk to us at any point! [The Fabric Discord](https://discord.gg/v6v4pMv) is a great place to ask questions and get input on your contributions.
+3. Fabric isn't easy to get your content into, as we uphold standards of quality and are very paranoid about making a bad precedent. You will be scrutinized again, and again, and again; you will be asked questions about more-or-less every line in your pull request (even the imports!).
+
+## Knowing When To Aim For Fabric
+
+The first step to contributing to Fabric API is coming up with an idea, and while all ideas may be useful, not all ideas necessarily fit within the scope of Fabric API.
+
+Fabric API tends to be restricted to small, highly used, and breaking changes to the vanilla game. Examples include helpers for adding blocks, which are used by basically every modder ever (hence their inclusion); breaking changes to the rendering system, which ensure that mods can remain compatible with one another; or commonly used mixins that, because of their nature, would otherwise cause mods to be incompatible (think redirects or overwrites). At the end of the day Fabric API exists for ease of implementing and modifying vanilla features, and compatibility between mods.
```

Comment:
Sentence here is really long and should probably be split up a bit.

I'd use `mod developer` over `modder` for a more professional sounding tone. I'd probably mention that features which are hard to implement correctly (such as networking api) are within scope.